### PR TITLE
Fix #4029 Ensure changeKernels always resolves, even in error states

### LINKED ISSUE #4029: Ensure changeKernels always resolves, even in error states
We've had a number of issues where the kernels/attach to dropdown can get stuck in error states.

With the current design, we are waiting on changeKernels to resolve in notebookModel before proceeding to populate the dropdowns.

Either change this design to deal with error conditions like rejecting, or ensure that we always resolve.

### PR DESCRIPTION
**Pull request by kevcunnane**
This is necessary to unblock reverting the kernel on canceling  Python install
- startSession now correctly sets up kernel information, since a kernel is loaded there.
- Remove call to change kernel on session initialize. This isn't needed due to refactor
- Handle kernel change failure by attempting to fall back to old kernel
- ExtensionHost $startNewSession now ensures errors are sent across the wire.
- Update AttachTo and Kernel dropdowns so they handle kernel being available. This is needed since other changes mean the session is likely ready before these get going

### CODE REVIEW

**Comment by kevcunnane** [comment]
This needs extensive testing before we consider for March, but given it unblocks key scenarios like cancel Python install not breaking the notebook we should take it if it proves stable.

File: src/sql/parts/notebook/models/clientSession.ts [hunk 1]
```diff
@@ -70,20 +70,22 @@ export class ClientSession implements IClientSession {
...
+		}
 	}
 
 	private async startServer(): Promise<void> {
 		let serverManager = this.notebookManager.serverManager;
 		if (serverManager && !serverManager.isStarted) {
 			await serverManager.startServer();
 			if (!serverManager.isStarted) {
-				throw new Error(nls.localize('ServerNotStarted', 'Server did not start for unknown reason'));
+				throw new Error(localize('ServerNotStarted', 'Server did not start for unknown reason'));
```
Comment (rajmusuku) [inline src/sql/parts/notebook/models/clientSession.ts]: Minor: Key should be in camel case => serverNotStarted

Comment (YurongHe) [reply src/sql/parts/notebook/models/clientSession.ts]: You told me to use double "" in localize msg:)

File: src/sql/parts/notebook/models/notebookModel.ts [hunk 2]
```diff
@@ -552,35 +566,68 @@ export class NotebookModel extends Disposable implements INotebookModel {
...
+			}
+			if (changeKernelNeeded) {
+				let spec = this.findSpec(displayName);
+				if (this._activeClientSession && this._activeClientSession.isReady) {
+					let kernel = await this._activeClientSession.changeKernel(spec, this._oldKernel);
+					try {
+						await kernel.ready;
+						await this.updateKernelInfoOnKernelChange(kernel);
+					} catch (err2) {
+						// TODO should we handle this in any way?
```
Comment (YurongHe) [inline src/sql/parts/notebook/models/notebookModel.ts]: Do you think console.log will help us for troubleshooting?

File: src/sql/parts/notebook/models/clientSession.ts (hunk 1 above)
Comment (kevcunnane) [reply src/sql/parts/notebook/models/clientSession.ts]: Fixed in next iteration, thanks 😄

File: src/sql/parts/notebook/models/notebookModel.ts (hunk 2 above)
Comment (kevcunnane) [reply src/sql/parts/notebook/models/notebookModel.ts]: Will add logging.

**Pull request merged**